### PR TITLE
Overriding config demo and PRODENG terraform plan

### PR DESCRIPTION
**Pull request by james-nesbitt**
- added terraform plan being used by prodeng 
- added new demo for overriding config (using new terraform plan)
- some cleanup in mtt modules and demo

### CODE REVIEW

File: demos/overriding_config/config/terraform.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    username: "{user:id?user}"
+    task_name: "{global:id}"
+    project: "{global:project}"
+    expire_duration: "72h"
+    kube_orchestration: true
+    dtr_install_flags: ["--ucp-insecure-tls"]
+    platform_repo: "public"
+    engine_version: "19.03.12"
+    ucp_version: "3.3.3"
+    dtr_version: "2.8.3"
```
Comment (quadespresso) [inline demos/overriding_config/config/terraform.yml:26]: `msr` not `dtr`?

**Review by quadespresso — CHANGES_REQUESTED** [review]
That was a great read!

My main concern has to do with when you cloned the Terraform code that I've been working on. You'll find that `dtr` has been changed to `msr`, including the module path and refs to it.

Other than that, just minor quibbles I think.

File: demos/overriding_config/config/terraform.yml (hunk 1 above)
Comment (james-nesbitt) [reply demos/overriding_config/config/terraform.yml:26]: these are values that go into the terraform plan tfvars file.  All references are effectively foreign keys that depend on the terraform plan

File: demos/overriding_config/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+This approach preaches some patterns but attempts to be readable enough that
+it can be interpreted for different approaches.
+
+## Cluster definition
+
+Different cluster definition folders contain config overrides that change the
+cluster definition.
+
+1. poc : man:1, worker:3
+2. small_business: man3, worker: 10
```
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/README.md:15]: this is a demo, meant to show ideas.  I can update the values though

File: demos/overriding_config/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+3. data_center : man:10, worker:100
```
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/README.md:16]: this is a demo, meant to show ideas.  I can update the values though

File: demos/overriding_config/config/cluster/data_center/terraform.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+vars:
+    manager_count: 10
```
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/config/cluster/data_center/terraform.yml:2]: these are values that go into the terraform plan tfvars file. All references are effectively foreign keys that depend on the terraform plan

File: demos/overriding_config/config/cluster/poc/terraform.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+vars:
+    manager_count: 1
+    worker_count: 3
+    dtr_count: 0
```
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/config/cluster/poc/terraform.yml:4]: these are values that go into the terraform plan tfvars file. All references are effectively foreign keys that depend on the terraform plan

File: demos/overriding_config/config/cluster/small_business/terraform.yml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+vars:
+    manager_count: 5
+    worker_count: 10
+    dtr_count: 3
```
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/config/cluster/small_business/terraform.yml:4]: these are values that go into the terraform plan tfvars file. All references are effectively foreign keys that depend on the terraform plan

File: demos/overriding_config/config/cluster/small_business/terraform.yml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+vars:
+    manager_count: 5
```
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/config/cluster/small_business/terraform.yml:2]: this is a demo, meant to show ideas.  I can update the values though

File: demos/overriding_config/config/cluster/small_business/terraform.yml [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+    worker_count: 10
```
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/config/cluster/small_business/terraform.yml:3]: this is a demo, meant to show ideas.  I can update the values though

File: demos/overriding_config/README.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
+# Overriding Configuration
+
+A demo on how to use the overriding/overlaiding features of the config handler
```
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/README.md:3]: obviously you've never been overlaid before

File: demos/overriding_config/config/cluster/data_center/terraform.yml [hunk 10]
```diff
@@ -0,0 +1,5 @@
...
+    worker_count: 100
+    dtr_count: 10
```
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/config/cluster/data_center/terraform.yml:4]: done

File: demos/overriding_config/README.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+to get adaptability in your test harness
```
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/README.md]: done

File: demos/overriding_config/README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/README.md]: fixed

File: demos/overriding_config/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/README.md]: fixed

File: demos/overriding_config/config/cluster/data_center/terraform.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/config/cluster/data_center/terraform.yml]: fixed

File: demos/overriding_config/config/cluster/small_business/terraform.yml (hunk 8 above)
[the remark this answers is not in the document]
Comment (james-nesbitt) [reply demos/overriding_config/config/cluster/small_business/terraform.yml]: fixed

**Pull request merged**